File: PubnativeLite/PubnativeLiteDemo/Analytics/AnalyticsDetailViewController.m
// 
// HyBid SDK License
//
// https://github.com/pubnative/pubnative-hybid-ios-sdk/blob/main/LICENSE
//

#import "AnalyticsDetailViewController.h"

@interface AnalyticsDetailViewController ()

@property (weak, nonatomic) IBOutlet UILabel *eventTypeLabel;
@property (weak, nonatomic) IBOutlet UITextView *eventJSONTextView;

@end

@implementation AnalyticsDetailViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.eventTypeLabel.text = self.event.eventType;
    self.eventJSONTextView.text = [self.event toJSON];
}

- (IBAction)dismissButtonTouchUpInside:(UIButton *)sender {
    [self dismissViewControllerAnimated:YES completion:nil];
}

@end

File: PubnativeLite/PubnativeLiteDemo/Analytics/AnalyticsDetailViewController.swift
// 
// HyBid SDK License
//
// https://github.com/pubnative/pubnative-hybid-ios-sdk/blob/main/LICENSE
//

class AnalyticsDetailViewController: UIViewController {

    @IBOutlet weak var eventTypeLabel: UILabel!
    @IBOutlet weak var eventJSONTextView: UITextView!
    
    var event: HyBidReportingEvent?

    override func viewDidLoad() {
        super.viewDidLoad()
        guard let event else { return }
        eventTypeLabel.text = event.eventType
        eventJSONTextView.text = event.propertiesValue()
    }

    @IBAction func dismissButtonTouchUpInside(_ sender: UIButton) {
        dismiss(animated: true, completion: nil)
    }
}


